pkg/pmem-device-manager/pmd-ndctl.go

```diff
@@ -97,16 +97,21 @@ func (pmem *pmemNdctl) CreateDevice(name string, size uint64, nsmode string) err
 		glog.V(4).Infof("Device with name: %s already exists, refuse to create another", name)
 		return fmt.Errorf("CreateDevice: Failed: namespace with that name exists")
 	}
-	// Increase to minimum size supported by libndctl.
-	const minsize uint64 = 2 * 1024 * 1024 * 1024
-	if size < minsize {
-		size = minsize
+	if size <= 0 {
+		// If called with zero size, we force small nonzero, otherwise size becomes equal to
+		// align after compensation, remains same after up-aligning and causes CreateNamespace to fail.
+		// In other words, combined compensate-and-align logic works for nonzero values.
+		glog.V(4).Infof("CreateDevice:%s: size %d is too low, adjusting to 4096", name, size)
+		size = 4096
```

Comment:
Why 4096 and not, say, 1?

Reply:
Why is that only relevant for size == 0 and not for size == align?

Can you document the behavior of libndctl or, if we don't know exactly, the observations that have been made about it? Perhaps then it becomes clearer why all of these size adjustments are made.

Reply:
> Why is that only relevant for size == 0 and not for size == align?

There will lead to different result:
size == 0
1. we compensate up by one align step --> size will be 1GB
2. align (inside CreateNamespace) leaves it same
3. libndctl creation of 1GB will fail

size == align
1. we compensate up by one align step --> size will be 2GB
2. align (inside CreateNamespace) leaves it same
3. libndctl creation of 2GB will create 1 GB namespace 

Thats why we have to force-set zero size to any small value to get:
size = 4096
1. we compensate up by one align step --> size will be 1GB + 4096bytes
2. align (inside CreateNamespace) aligns it up to 2 GB
3. libndctl creation of 2GB will create 1 GB namespace

It's true that exactly 4096 is not needed, 1 would do it same way (should perhaps add comment).
I thought that 1 is too weird value for volume size.

Reply:
If the goal is to be larger than zero, then 1 is closer to the goal than 4096. But that' subjective and both need to be explained.